### PR TITLE
[bibdata] Move dependencies into playbook to reduce repetition

### PR DESCRIPTION
**Pull request by maxkadel**
- Move role dependencies for testing to converge.yml
- List all roles that are actually needed, comment out roles that are currently duplicates

~As a nice side effect, test runs appear to be [much faster on average](https://github.com/pulibrary/princeton_ansible/actions/workflows/molecule_tests.yml?query=branch%3Abibdata_move_roles_to_playbook+is%3Asuccess)~

To verify, switch to fresh staging VMs, run the playbook, deploy bibdata to the fresh staging VMs, and trigger an incremental update on the Alma sandbox.

Performed the above on April 9-10. 2024, and successfully built the boxes and ran jobs on Sidekiq.

### CODE REVIEW

File: playbooks/bibdata.yml [hunk 1]
```diff
@@ -12,12 +12,31 @@
...
-    - role: roles/bibdata
-    - role: roles/bibdata_sqs_poller
-    - role: 'hr_share'
-    - role: roles/datadog
+    - role: redis
+    # re-add when removed from passenger
+    # - role: ruby_s
+    # re-add when removed from rails_app
+    # - role: bind9
+    # re-add when removed from rails_app
```
Comment (sandbergja) [inline playbooks/bibdata.yml:23]: This is nitpicky, but I found it hard to know at a glance which role each comment applied to.  Would it be possible to add them to the comment?
```suggestion
    # re-add when passenger role is removed from rails_app
```

**Review by sandbergja — COMMENTED** [review]
Thanks, @maxkadel !  I think it's great, just one suggestion for comment readability

File: playbooks/bibdata.yml (hunk 1 above)
Comment (maxkadel) [reply playbooks/bibdata.yml]: Thanks @sandbergja! Good idea, I've tried to do that

**Review by sandbergja — APPROVED** [review]
This is awesome, thanks so much @maxkadel 🦄

**Pull request merged**